server/cgroup_config.go

```
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path"
+	"syscall"
+
+	"github.com/tumdum/jobserver"
+)
+
+const (
+	childCgroupPrefix = "job_server_child"
+
+	rootCGEnvVar      = "JOB_SERVER_ROOT_CG"
+	cpuLimitEnvVar    = "JOB_SERVER_CPU_LIMIT"
+	memoryLimitEnvVar = "JOB_SERVER_MEMORY_LIMIT"
+	ioLimitEnvVar     = "JOB_SERVER_IO_LIMIT"
+)
+
+type cgroupConfig struct {
+	rootCG string
+	cpu    string
+	mem    string
+	io     string
+}
+
+func (c *cgroupConfig) isEnabled() bool {
+	return len(c.rootCG) > 0 && (len(c.cpu) > 0 || len(c.mem) > 0 || len(c.io) > 0)
+}
+
+func (c *cgroupConfig) setUpCgroups(jobID jobserver.JobID, pid int) {
+	if !c.isEnabled() {
+		log.Println("CGroups will not be set up")
+		return
+	}
+	log.Printf("Will create cgroups under '%v'", c.rootCG)
+	subCG, err := c.addToCgroup(jobID, pid)
+	if err != nil {
+		log.Println(err)
+		return
+	}
+
+	// Will not fail when cgroups operations fail. This makes it easier for
+	// testing cgroup code. In real a production environment those errors should
+	// prevent running of job.
+	c.writeToFile(subCG, "cpu.max", c.cpu)
+	c.writeToFile(subCG, "memory.max", c.mem)
+	c.writeToFile(subCG, "io.max", c.io)
```

Do not ignore the errors.

---

Fixed in 98f1172ee7471077046f70ad9c3068c41c31c52b